metal/devices.go

```diff
@@ -60,13 +61,19 @@ func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloud
 	if err != nil {
 		return nil, err
 	}
-	nodeAddresses, err := nodeAddresses(device)
+	// was a node IP provided
+	var providedNodeIP string
+	if node.Annotations != nil {
+		providedNodeIP = node.Annotations[cpapi.AnnotationAlphaProvidedIPAddr]
+	}
+	nodeAddresses, err := nodeAddresses(device, providedNodeIP)
```

Comment:
I think we would get this for free by starting with `nodeAddresses := append([]v1.NodeAddress, node.Status.Addresses...)` (however it is we get the existing addresses).

If we don't use the existing node addresses, we may omit addresses manually defined or defined through other kubelet options. For example `--hostname-override`:
* https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetesiohostname
* https://kubernetes.io/docs/concepts/architecture/nodes/#addresses

Reply:
There are no other structures to provide addresses to the kubelet other than `--node-ip` and `--hostname-override`. This PR addresses `--node-ip`, but I think you are  correct to raise that it does _not_ resolve the issue of `--hostname-override`. It actually explicitly sets the hostname based on the response from the EQXM API.

I am going to leave this one for now. It isn't quite so simple. There are 3 distinct hostnames to worry about:

*  whatever the kubelet got from the node automatically (by running `hostname`); this might to might not be what is in the EQXM API, e.g. the user changed the hostname via command manually, userdata script, etc.
* `--hostname-override`
* EQXM API

Reconciling all three of those is its own logical mess. So I am leaving it out of this PR. This PR doesn't change the logic, however flawed: hostname is set via EQXM API.

We can set a new one for hostname after this one is done.